prime-router/src/main/kotlin/fhirengine/utils/HL7Reader.kt

```diff
@@ -91,5 +92,40 @@ class HL7Reader(private val actionLogger: ActionLogger) : Logging {
                 timestamp.ts1_Time.valueAsDate
             } else null
         }
+
+        fun getMessageType(message: Message): String? {
+            return (message["MSH"] as MSH).msh9_MessageType.msg1_MessageCode.toString()
+        }
+
+        fun isBirthTime(message: Message): Boolean {
+            val patient = getORMPatient(message) ?: return false
+            val dateTimeOfBirth = patient.pid.dateTimeOfBirth ?: return false
+            return dateTimeOfBirth.ts1_Time.toString().length > 8
```

Comment:
It might be easier to get the birth time using the Terser and avoid having to get the patient and have all these checks in place.
I think this should work:
```
val pid = message.get("PID") as Segment
val dateTimeOfBirth = Terser.get(pid, 7, 0, 1, 1)
```

Reply:
Cool that cleans things up quite a bit, thank you!

Reply:
So, with ORM it is not always PID which is why it is necessary to check for either the PATIENT or the PATIENT_RESULT.

Reply:
So I tried what we discussed this morning and `val pid = message.get("PATIENT") as Segment` results in `class ca.uhn.hl7v2.model.v251.group.ORM_O01_PATIENT cannot be cast to class ca.uhn.hl7v2.model.Segment ` so basically it doesn't eliminate the problem you were trying to solve because it will still need to differentiate between the message types.